src/blocks/weather.rs

```diff
@@ -141,6 +145,35 @@ impl Weather {
                     .ok_or_else(malformed_json_error)?;
 
                 // Convert wind direction in azimuth degrees to abbreviation names
+                // Compute the felt temperature according to Australian
+                // apparent temperature as found on wikipedia
+                // unsure about how to properly make if statement
+                let check = match *units {
```

Comment:
A more idiomatic approach in Rust is to use `match` statements whenever possible, instead of `if`. I would suggest moving everything into one `match` statement here:

```rust
let apparent_temp = match *units {
  OpenWeatherMapUnits::Metric => {
    // All calculations for metric units here.
  },
  OpenWeatherMapUnits::Imperial => {
    // All calculations for imperial units here.
  },
};
```

This will mean some duplication of code, but it will make it much easier to understand what is happening.

Reply:
I structured the code the way I did because the formula is only published for metric units. So I figured it would make sense to do all the calculations in metric and then just do unit conversions for "imperial" units.

Reply:
Yeah I guess it's OK for now since I didn't find an imperial formula from a quick Google.